dashboard/app/controllers/followers_controller.rb

```diff
@@ -26,10 +25,14 @@ def student_register
     end
 
     Retryable.retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
-      if @user.save && @section&.add_student(@user)
-        sign_in(:user, @user)
-        redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name)
-        return
+      if !verify_recaptcha
```

Comment:
I'm not very familiar with this code, but it seems a little odd to me that the captcha verification would be inside the retryable block?  Maybe move this outside the retryable block?